src/core/thread/key_manager.cpp

```diff
@@ -97,6 +97,22 @@ void KeyManager::Stop(void)
 }
 
 #if OPENTHREAD_MTD || OPENTHREAD_FTD
+bool KeyManager::IsPSKcSet(void) const
+{
+    bool isSet = false;
+
+    for (uint8_t i = 0; i < sizeof(mPSKc); i++)
+    {
+        if (mPSKc.m8[i])
+        {
+            ExitNow(isSet = true);
+        }
+    }
```

Comment:
How about adding a boolean member variable to track if it's set or not? `KeyManager` already has odd size (due to `uint8_t mSecurityPolicyFlags` at the end) so we can probably get an extra  byte/bool for free - due to padding.

Reply:
I think the real question is whether we want all-zeros to be a valid value. Thoughts?

Reply:
I would say it'd be good to not add any restrictions (allow all-zeros as a valid value) if we can.

Reply:
Is it possible we generate random value in constructor of KeyManager no matter it's zero or not?

Reply:
Given that the current default is not a random value, I don't think we want to enable Thread Commissioning if the PSKc is set to all zeros. In that case, we should discourage configuring the PSKc to an all-zeros value.

Reply:
@bukepo, one issue with the constructor is that `otPlatRandomGetTrue()` can return an error, which we cannot propagate back to the caller in a constructor.

Reply:
Yes, that is a problem, can we have a Init method for KeyManager? Alternatively If we add a flag indicating PSKc set or not, we may consider only start border agent service when set.

Reply:
@bukepo, of course we can add a separate Init method. The point is that we should not enable border agent if the PSKc is set to all zeros. Whether the PSKc was locally set is not relevant, given that the PSKc may be received via an existing Thread network.

Reply:
I updated this PR to use a flag instead. We can discuss separately how to implement the border agent policy.